src/projects/shared/api/circuit/tests/public/Component.test.ts

```diff
@@ -11,66 +11,295 @@ import {CreateTestRootCircuit} from "tests/helpers/CreateTestCircuit";
+        });
+
+        test("getProp with invalid prop", () => {
+            const [circuit, _, {PlaceAt, Connect, GetPort}] = CreateTestRootCircuit();
+            const [c] = PlaceAt(V(0, 0));
+            expect(c.getProp("fakeProp")).toBeUndefined();
+        });
+
+        test("Set invalid prop", () => {
+            const [circuit, _, {PlaceAt, Connect, GetPort}] = CreateTestRootCircuit();
+            const [c] = PlaceAt(V(0, 0));
+            expect(() => {c.setProp("fakeProp", true)}).toThrow();
+        });
     });
 
     describe("Transform", () => {
-        // Test ideas:
-        // Making sure default x/y/pos/angle are 0s
-        // Setting positions => getting position
-        // Undo/redo settings positions
+        test("Basic movement", () => {
+            const [circuit, _, {PlaceAt, Connect, GetPort}] = CreateTestRootCircuit();
+            const [c] = PlaceAt(V(0, 0));
+            expect(c.x).toBe(0);
+            expect(c.y).toBe(0);
+            expect(c.pos.x).toBe(0);
+            expect(c.pos.y).toBe(0);
+            expect(c.angle).toBe(0);
+
+            c.x = 3;
+            expect(c.x).toBe(3);
+            expect(c.y).toBe(0);
+            expect(c.pos.x).toBe(3);
+            expect(c.pos.y).toBe(0);
+            expect(c.angle).toBe(0);
+            c.y = 5;
+            expect(c.x).toBe(3);
+            expect(c.y).toBe(5);
+            expect(c.pos.x).toBe(3);
+            expect(c.pos.y).toBe(5);
+            expect(c.angle).toBe(0);
+
+            circuit.undo();
+            expect(c.x).toBe(3);
+            expect(c.y).toBe(0);
+            expect(c.pos.x).toBe(3);
+            expect(c.pos.y).toBe(0);
+            expect(c.angle).toBe(0);
+            circuit.redo();
+            expect(c.x).toBe(3);
+            expect(c.y).toBe(5);
+            expect(c.pos.x).toBe(3);
+            expect(c.pos.y).toBe(5);
+            expect(c.angle).toBe(0);
+        });
+
+        test("Movement with .pos", () => {
+            const [circuit, _, {PlaceAt}] = CreateTestRootCircuit();
+            const [c] = PlaceAt(V(0, 0));
+            expect(c.x).toBe(0);
+            expect(c.y).toBe(0);
+            expect(c.pos.x).toBe(0);
+            expect(c.pos.y).toBe(0);
+            expect(c.angle).toBe(0);
+
+            c.pos = V(9, -17);
+            expect(c.x).toBe(9);
+            expect(c.y).toBe(-17);
+            expect(c.pos.x).toBe(9);
+            expect(c.pos.y).toBe(-17);
+            expect(c.angle).toBe(0);
+
+            // undo/redo
+            circuit.undo();
+            expect(c.x).toBe(0);
+            expect(c.y).toBe(0);
+            expect(c.pos.x).toBe(0);
+            expect(c.pos.y).toBe(0);
+            expect(c.angle).toBe(0);
+            circuit.redo();
+            expect(c.x).toBe(9);
+            expect(c.y).toBe(-17);
+            expect(c.pos.x).toBe(9);
+            expect(c.pos.y).toBe(-17);
+            expect(c.angle).toBe(0);
+        });
```

Comment:
Would be good to have some tests where you do multiple moves in a single transaction, i.e.
```
c.pos = V(0, 0);
circuit.beginTransaction();
c.pos = V(5, 0);
c.pos = V(10, 0);
circuit.commitTransaction();
expect(c.pos).toEqual(V(10, 0));
circuit.undo();
expect(c.pos).toEqual(V(0, 0)); // ignores the V(5, 0)
```